### PR TITLE
Fix/Refactor MessageInputView

### PR DESCRIPTION
**Pull request by adrian09h**
# Submit a pull request

- clean up `MessageInputView`

### CODE REVIEW

File: library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java [hunk 1]
```diff
@@ -146,46 +156,100 @@ private void applyStyle() {
...
+                permissionRequestListener.openPermissionRequest();
+        });
+    }
+
+    private void configInputEditText(){
+        binding.etMessage.setOnFocusChangeListener((View view, boolean hasFocus)-> {
+            viewModel.setInputType(hasFocus ? InputType.SELECT : InputType.DEFAULT);
+            if (hasFocus) {
+                lockScrollUp = true;
+                new Handler().postDelayed(() -> lockScrollUp = false, 500);
```
Comment (elevenetc) [inline library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java]: No need to create new `Handler`. Each view has it already and delayed action could be executed with `view.postDelayed`.

File: library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java [hunk 2]
```diff
@@ -146,46 +156,100 @@ private void applyStyle() {
...
+                Utils.showSoftKeyboard((Activity) getContext());
+            } else
+                Utils.hideSoftKeyboard((Activity) getContext());
+        });
+
+        binding.etMessage.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                //noop
```
Comment (elevenetc) [inline library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java]: If you want to use just one method of an interface (`TextWatcher` in this case)  there is a way to make a little bit nicer for reading and save some space:
```java
// imported static method and lambda
afterTextChanged(binding.etMessage, editable -> {
      int l = editable.length();
      l++;
});
```
With functional interfaces and static utility class:
```java
interface Actions {
    interface EditableAction{
        void act(Editable editable);
    }
}

static class TestUtils {
    static void afterTextChanged(TextView textView, Actions.EditableAction action){
        textView.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
                //ignore
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                //ignore
            }

            @Override
            public void afterTextChanged(Editable editable) {
                action.act(editable);
            }
        });
    }
}
```

File: library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java [hunk 3]
```diff
@@ -146,46 +156,100 @@ private void applyStyle() {
...
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
```
Comment (elevenetc) [inline library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java]: When you override interface methods try to also fix names if IDE didn't do it for you:
```
CharSequence s, int start, int before, int count
```
Even for unused parameters it's a readability bonus, plus you might save some time for a future developer who might want to use it.

Comment (adrian09h) [reply library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java]: @elevenetc updated everything according to your comments.

**Comment by adrian09h** [comment]
@elevenetc resolved conflicts

File: library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java [hunk 4]
```diff
@@ -291,53 +477,10 @@ private void editMessage(Message message) {
...
-            if (this.openCameraViewListener != null)
-                openCameraViewListener.openCameraView(chooserIntent, Constant.CAPTURE_IMAGE_REQUEST_CODE);
-        });
-        binding.llFile.setOnClickListener(v -> messageInputController.onClickOpenSelectFileView(v, null));
-    }
-
     // endregion
+
+    // region permission check
+    // TODO: the name of this method is weird (progres..)? perhaps captureMedia?
```
Comment (elevenetc) [inline library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java]: Seems `TODO` might be removed.

Comment (adrian09h) [reply library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java]: I have removed this in the other PR so it will be removed soon.

**Pull request merged**